pyproject.toml

```diff
@@ -35,7 +35,8 @@ dependencies = [
     "azure-mgmt-compute==30.1.0",
     "azure-mgmt-resource==23.0.1",
     "azure-cli-core==2.50.0",
-    "platformdirs"
+    "platformdirs",
+    "python-dateutil==2.9",
```

Comment:
I suspect this pin will cause some pain fyi

Reply:
Is this internal pain or for our customers?

Reply:
our customers, strict pins like this are somewhat painful.

Reply:
Ya agreed - lets figure out a good plan after this pr gets out to address the pinned versions.

Reply:
Made a ticket to track that: https://synccomputing.atlassian.net/browse/PROD-2324

Can bring it up at parking lot, would pinning as <3 be safe as that's the next major version that might introduce breaks?